src/main/java/org/jfrog/hudson/util/converters/GenericDeployerResolverOverriderConverter.java

```diff
@@ -0,0 +1,114 @@
+                oldDeployerDetails = oldResolverDetailsField.get(overrider);
+            }
+
+            if (oldDeployerDetails == null || oldResolverDetailsField.get(overrider) == null) {
+                // No old fields found, therefore convert is not needed
+                return;
+            }
+
+            Field specsResolverDetailsField = overriderClass.getDeclaredField("specsResolverDetails");
+            Field legacyResolverDetailsField = overriderClass.getDeclaredField("legacyResolverDetails");
+            ServerDetails resolverServerDetails = createInitialResolveDetailsFromDeployDetails((ServerDetails) oldDeployerDetails);
+            setServerDetailsIfNew(specsResolverDetailsField, overrider, resolverServerDetails);
+            setServerDetailsIfNew(legacyResolverDetailsField, overrider, resolverServerDetails);
+        } catch (ReflectiveOperationException e) {
+            converterErrors.add(getConversionErrorMessage(overrider, e));
+        }
+    }
+
+    /**
+     * Convert deployerDetails -> details -> specsDeployerDetails/legacyDeployerDetails
+     * Convert the details or deployerDetails to specsDeployerDetails or legacyDeployerDetails if they doesn't exists already
+     * This conversion comes after a name change (deployerDetails -> specsDeployerDetails and legacyDeployerDetails)
+     */
+    @Override
+    void overrideDeployerDetails(DeployerOverrider overrider, Class<?> overriderClass) {
+        try {
+            Field oldDeployerDetailsField = overriderClass.getDeclaredField("deployerDetails");
+            oldDeployerDetailsField.setAccessible(true);
+            Object oldDeployerDetails = oldDeployerDetailsField.get(overrider);
+            if (oldDeployerDetails == null) {
+                oldDeployerDetailsField = overrider.getClass().getDeclaredField("details");
+                oldDeployerDetailsField.setAccessible(true);
+                oldDeployerDetails = oldDeployerDetailsField.get(overrider);
+            }
+
+            if (oldDeployerDetails == null || oldDeployerDetailsField.get(overrider) == null) {
+                // No old fields found, therefore convert is not needed
+                return;
+            }
+
+            Field specsDeployerDetailsField = overriderClass.getDeclaredField("specsDeployerDetails");
+            Field legacyDeployerDetailsField = overriderClass.getDeclaredField("legacyDeployerDetails");
+            ServerDetails deployerServerDetails = createInitialDeployDetailsFromOldDeployDetails((ServerDetails) oldDeployerDetails);
+            setServerDetailsIfNew(specsDeployerDetailsField, overrider, deployerServerDetails);
+            setServerDetailsIfNew(legacyDeployerDetailsField, overrider, deployerServerDetails);
+        } catch (ReflectiveOperationException e) {
+            converterErrors.add(getConversionErrorMessage(overrider, e));
+        }
+    }
+
+    /**
+     * Set server details if not already exist.
```

Comment:
Set server details if not already exist. --> Set the server details, if they don't exist already.